src/layers/CementLayer.ts

```diff
@@ -60,9 +60,21 @@ export class CementLayer extends WellboreBaseComponentLayer {
       const points = [];
       let prevAngle = 10000;
       const allowedAngleDiff = 0.0005;
+      const morePointsForStartAndEndMeters = 10;
+
+      // Always add first points
+
+      for (let i = c.toc; i < c.toc + morePointsForStartAndEndMeters; i += this.options.wellboreBaseComponentIncrement) {
+        const p = this.referenceSystem.project(i);
+        points.push({ point: new Point(p[0], p[1]), md: i });
```

Comment:
should validate that `i` is not negative or clamp the value when passing it into list of points.